commands/update_organization_setcredentials.go

```diff
@@ -0,0 +1,349 @@
+			return microerror.Mask(err)
+		}
+
+		foundOrg := false
+		for _, org := range orgsResponse.Payload {
+			if org.ID == args.organizationID {
+				foundOrg = true
+			}
+		}
+		if !foundOrg {
+			return microerror.Mask(organizationNotFoundError)
+		}
+	}
+
+	return nil
+}
+
+// updateOrgSetCredentialsRunOutput calls the busniness function and produces
+// meanigful terminal output.
+func updateOrgSetCredentialsRunOutput(cmd *cobra.Command, cmdLineArgs []string) {
+	args := defaultUpdateOrgSetCredentialsArguments()
+	result, err := updateOrgSetCredentials(args)
+
+	if err != nil {
+		handleCommonErrors(err)
+
+		// From here on we handle errors that can only occur in this command
+		headline := ""
+		subtext := ""
+
+		switch {
+		case err.Error() == "":
+			return
+		case IsCredentialsAlreadySetError(err):
+			headline = "Credentials already set"
+			subtext = fmt.Sprintf("Organization '%s' has credentials already. These cannot be overwritten.", args.organizationID)
+		default:
+			headline = err.Error()
+		}
+
+		// print output
+		fmt.Println(color.RedString(headline))
+		if subtext != "" {
+			fmt.Println(subtext)
+		}
+		os.Exit(1)
+	}
+
+	// success
+	fmt.Println(color.GreenString("Credentials set successfully"))
+	fmt.Printf("The credentials are stored with the unique ID '%s'.\n", result.credentialID)
+}
+
+// updateOrgSetCredentials performs the API call and provides a result.
+func updateOrgSetCredentials(args updateOrgSetCredentialsArguments) (*updateOrgSetCredentialsResult, error) {
+	// build request body based on provider
+	requestBody := &models.V4AddCredentialsRequest{Provider: &provider}
+	if provider == "aws" {
+		requestBody.Aws = &models.V4AddCredentialsRequestAws{
+			Roles: &models.V4AddCredentialsRequestAwsRoles{
+				Admin:       &args.awsAdminRole,
+				Awsoperator: &args.awsOperatorRole,
+			},
+		}
+	} else if provider == "azure" {
+		requestBody.Azure = &models.V4AddCredentialsRequestAzure{
+			Credential: &models.V4AddCredentialsRequestAzureCredential{
+				SubscriptionID: &args.azureSubscriptionID,
+				TenantID:       &args.azureTenantID,
+				ClientID:       &args.azureClientID,
+				SecretKey:      &args.azureSecretKey,
+			},
+		}
+	}
+
+	if args.verbose {
+		fmt.Println(color.WhiteString("Sending API request to set credentials"))
+	}
+	auxParams := ClientV2.DefaultAuxiliaryParams()
+	auxParams.ActivityName = updateOrgSetCredentialsActivityName
+	response, err := ClientV2.SetCredentials(args.organizationID, requestBody, auxParams)
+	if err != nil {
+		if clientErr, ok := err.(*clienterror.APIError); ok {
+			if clientErr.HTTPStatusCode == http.StatusConflict {
+				return nil, microerror.Mask(credentialsAlreadySetError)
+			}
+		}
+		return nil, microerror.Mask(err)
+	}
+
+	// Location header returned is in the format
+	// /v4/organizations/myorg/credentials/{credential_id}/
+	segments := strings.Split(response.Location, "/")
+	result := &updateOrgSetCredentialsResult{
+		credentialID: segments[len(segments)-2],
```

Comment:
What is this magical **2** here for ?

Reply:
It's to return the second last item from the slice. See https://play.golang.org/p/krFhD0fU_K-